### ISSUE TITLE
Link to schema broken when requestBody entity name is not same as schema name

### ISSUE THREAD

**Issue by gwittel** [Mermade/widdershins#236]
**Describe the bug**

* [x] - I have checked that my input document is valid OpenAPI 2.0/3.0.x or AsyncAPI 1.x

Using the petstore sample as a starting point, if you have an OpenAPI doc that has a structure like:

```yaml
    post:
      tags:
        - pet
      # ....
       requestBody:
        $ref: '#/components/requestBodies/PetRequest'
# ...
  requestBodies:
    PetRequest:
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/Pet'
# ...
  schemas:
    Pet:
      type: object
      # ....
```

The top level link from the generated 'pet' resource's `addPet` body parameter `PetRequest` will be broken.  It seems that the request body item name (`PetRequest`) must match the schema name (`Pet`).  The default petstore example happens to share the same request body name with the schema.  In cases like the above, the link won't work.

Broken case:  The link points to `#schemapetrequest` anchor in the generated markdown.

**To Reproduce**

1. Command-line:
   ```
    # See linked gist below.
    $ widdershins petstore3.mod.json -o petstore3.mod.md
    ```
2. My input file: https://gist.github.com/gwittel/335e1f7685251cabcb7e2f850cd1d0df
3. One can see links to the request body. But no such anchor is defined
    ```
    $ grep 'schemapetrequest' petstore3.mod.md
    |body|body|[PetRequest](#schemapetrequest)|true|Pet object that needs to be added to the store|
    |body|body|[PetRequest](#schemapetrequest)|true|Pet object that needs to be added to the store|
    $
    ```
4. Its a broken link

**Expected behavior**

The generated link to `#schemapetrequest` should point to an existing anchor OR be resolved to the underlying schema's anchor `#schemapet`

**Screenshots**

**Versions**:
- Node.js (note only LTS versions are supported):  v10.16.0
- Widdershins: 3.6.7

**Additional context**

**Comment by MikeRalphson** [comment]
Should be fixed in version `4.0.0`, which includes breaking changes (please see the `README.md` file).

**Comment by mushroomsir** [comment]
@MikeRalphson @gwittel  The version 4.0.1 is still unresolved (please reopen this issue).
